### PR TITLE
[exporter/datadog]: add env and tag name normalization to trace payloads

### PR DESCRIPTION
**Pull request by ericmustin**
**Description:** <Describe what has changed.>

This PR adds additional normalization to the datadogexporter for ensuring that the `env` output matches what the [`datadogagent` output would be](https://github.com/DataDog/datadog-agent/blob/c87e93a75b1fc97f0691faf78ae8eb2c280d6f55/pkg/trace/traceutil/normalize.go#L89). Specifically, we want to ensure that envs are lowercased. Additionally, we also want to now normalize tag key's.

**Link to tracking Issue:** <Issue number if applicable> this went stale so reopening https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2818

**Testing:** <Describe what testing was performed and which tests were added.> Added unit tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by ericmustin** [comment]
@mx-psi tried to address ur feedback from #2818 , lmk what u think

File: exporter/datadogexporter/translate_traces_test.go [hunk 1]
```diff
@@ -1058,3 +1060,91 @@ func TestStatsAggregations(t *testing.T) {
...
+	aggregatedTraces := aggregateTracePayloadsByEnv(outputTraces)
+
+	obfuscator := obfuscate.NewObfuscator(obfuscatorConfig)
+	obfuscatePayload(obfuscator, aggregatedTraces)
+	assert.Equal(t, 1, len(aggregatedTraces))
+
+	assert.Equal(t, "uppercase", aggregatedTraces[0].Env)
+}
+
+func TestNormalizeTag(t *testing.T) {
```
Comment (mx-psi) [inline exporter/datadogexporter/translate_traces_test.go:1092]: Same thing as in the other PR, I think codecov doesn't like that this is in a different package. Can you move it to the `utils` package?

File: exporter/datadogexporter/translate_traces_test.go [hunk 2]
```diff
@@ -103,12 +105,12 @@ func NewResourceSpansData(mockTraceID [16]byte, mockSpanID [8]byte, mockParentSp
...
-			"namespace":                      pdata.NewAttributeValueString("kube-system"),
-			"service.name":                   pdata.NewAttributeValueString("test-resource-service-name"),
-			"deployment.environment":         pdata.NewAttributeValueString("test-env"),
-			"service.version":                pdata.NewAttributeValueString("test-version"),
+			conventions.AttributeContainerID:           pdata.NewAttributeValueString("3249847017410247"),
+			conventions.AttributeDeploymentEnvironment: pdata.NewAttributeValueString("test-env"),
+			conventions.AttributeK8sPod:                pdata.NewAttributeValueString("example-pod-name"),
+			"namespace":                                pdata.NewAttributeValueString("kube-system"),
+			"service.name":                             pdata.NewAttributeValueString("test-resource-service-name"),
+			"service.version":                          pdata.NewAttributeValueString("test-version"),
```
Comment (mx-psi) [inline exporter/datadogexporter/translate_traces_test.go:113]: nit: use `conventions` constants here, it helps to see which ones are OTel specific and which are not

**Review by mx-psi — APPROVED** [review]

**Comment by ericmustin** [comment]
@anuraaga could i get a review here when u have a moment? 🙇

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**